plone/app/discussion/browser/moderation.py

```diff
@@ -13,69 +15,119 @@
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from Products.statusmessages.interfaces import IStatusMessage
 from zope.event import notify
-from zope.interface import alsoProvides
+
+
+# Translations for generated values in buttons
+# States
+_('comment_pending', default='pending')
+# _('comment_approved', default='published')
+_('comment_published', default='published')
+_('comment_rejected', default='rejected')
+_('comment_spam', default='marked as spam')
+# Transitions
+_('Recall')
+_('Approve')
+_('Reject')
+_('Spam')
+
+
+class TranslationHelper(BrowserView):
```

Comment:
I am not really sure this is needed. I would avoid introducing this new view here.
AFAICS this is used only in `CommentTransition`. 
Would it be possible to move those methods there or you need this as an entry point to provide custom translations?